PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/component/ResourceGroupPanel.java

```diff
@@ -106,23 +104,19 @@ public void addItemListener(ItemListener actionListener) {
     private void createResourceGroup() {
         cbResourceGroup.setSelectedItem(null);
         cbResourceGroup.setPopupVisible(false);
-        final NewResourceGroupDialog dialog = new NewResourceGroupDialog();
+        final NewResourceGroupDialog dialog = new NewResourceGroupDialog(subscriptionId);
         dialog.pack();
-        dialog.setLocationRelativeTo(this);
-        dialog.addWindowListener(new WindowAdapter() {
-            @Override
-            public void windowClosed(WindowEvent windowEvent) {
-                super.windowClosed(windowEvent);
-                final ResourceGroupWrapper newResourceGroup = dialog.getResourceGroup();
-                if (newResourceGroup != null) {
-                    cbResourceGroup.addItem(newResourceGroup);
-                    cbResourceGroup.setSelectedItem(newResourceGroup);
-                } else {
-                    cbResourceGroup.setSelectedItem(selectedResourceGroup);
-                }
+        if (dialog.showAndGet()) {
+            final ResourceGroupWrapper newResourceGroup = dialog.getResourceGroup();
+            if (newResourceGroup != null) {
```

Comment:
shall we check whether there is already a newly create resource group with the same name(to avoid same new resource group name in one combobox)

Reply:
Nice catch!